airflow/models/taskinstance.py

```diff
@@ -1183,6 +1188,24 @@ def signal_handler(signum, frame):  # pylint: disable=unused-argument
         # Run on_execute callback
         self._run_execute_callback(context, task)
 
+        if task_copy.is_smart_sensor_compatible():
+            try:
+                # Must use task_copy instead of task otherwise no rendered value
+                if task_copy.register_in_sensor_service(self, context):
+                    self.log.info('Submitting %s to sensor service', self)
+                    self.state = State.SENSING
+                    self.start_date = timezone.utcnow()
+                    session.merge(self)
+                    session.commit()
+                    # Raise exception for sensing state
+                    raise AirflowSmartSensorException("Task successfully registered in smart sensor.")
+            except Exception as e:
+                if isinstance(e, AirflowSmartSensorException):
+                    raise e
+                self.log.warning("Failed to register in sensor service."
+                                 "Continue to run task in non smart sensor mode.")
+                self.log.exception(e, exc_info=True)
+
```

Comment:
Should it be possible to simplify this part?
```python
try:
    registered = task_copy.register_in_sensor_service(self, context)
except Exception as e:
    self.log.warning("Failed to register in sensor service."
                                  "Continue to run task in non smart sensor mode.")
    self.log.exception(e, exc_info=True)

if registered:
    self.log.info('Submitting %s to sensor service', self)
    self.state = State.SENSING
    self.start_date = timezone.utcnow()
    session.merge(self)
    session.commit()
    # Raise exception for sensing state
    raise AirflowSmartSensorException("Task successfully registered in smart 
```
WDYT? The last part can be encapsulated into a helper method

Reply:
Good point. Changed.